### PR TITLE
feat(controller): allow setting the Experiment's service name; don't hardcode ports

### PR DESCRIPTION
**Pull request by alexef**
Fixes: #2233 

Stop hardcoding service port.
Allow user to set the experiment service name.
Create a service even though a weight is not set (when the name is provided)

Example:
```
...
steps:
- experiment:
     templates: 
     - name: experiment-baseline
       specRef: stable
       service: 
          name: my-experiment # optional
          ports:
          - port: 80
            targetPort: 8080
            protocol: TCP
```

This is a follow up of https://github.com/argoproj/argo-rollouts/pull/2356, see also [this comment](https://github.com/argoproj/argo-rollouts/issues/2233#issuecomment-1286900573) for context.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/) with a list of types and scopes found [here](https://github.com/argoproj/argo-rollouts/blob/master/.github/workflows/pr-title-check.yml), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

**Comment by alexef** [comment]
@zachaller can you have a look at this?

File: experiments/service.go [hunk 1]
```diff
@@ -73,12 +72,8 @@ func (ec *experimentContext) CreateService(serviceName string, template v1alpha1
...
 		},
 		Spec: corev1.ServiceSpec{
-			Selector: selector,
-			Ports: []corev1.ServicePort{{
-				Protocol:   "TCP",
-				Port:       int32(80),
-				TargetPort: intstr.FromInt(8080),
-			}},
+			Selector: template.Service.Selector,
+			Ports:    template.Service.Ports,
```
Comment (zachaller) [inline experiments/service.go:76]: Why are we not setting the Spec whole templates.Service here?

Comment (alexef) [reply experiments/service.go:76]: Exactly, I couldn’t figure out a way to do it. Happy to learn

**Review by zachaller — APPROVED** [review]

**Comment by zachaller** [comment]
One last thing then I think this is good could you add some documentation and update docs about when a service is generated etc I think mainly around this page https://argoproj.github.io/argo-rollouts/features/experiment/ and maybe on the spec https://argoproj.github.io/argo-rollouts/features/specification/

**Comment by alexef** [comment]
Updated the docs, not sure if it's okay: https://github.com/argoproj/argo-rollouts/pull/2357/commits/0f5ed7afb764354148e3842d923e348dfb2f696a

Regarding service spec, we should probably also mark this in the docs as available from next release?

**Comment by alexef** [comment]
superseeded by: https://github.com/argoproj/argo-rollouts/pull/2397

**Pull request closed without merging**